### DISCUSSION TITLE
No Variable Refresh Rate when MonitorControl is running

### DISCUSSION THREAD

**Discussion by Hanage999** [MonitorControl/MonitorControl#1166] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

When the app is running, the refresh rate is kept at 144Hz even after you configure the monitor and the OS to enable Variable Refresh Rate (40Hz-144Hz).

### Steps to reproduce

1. Make sure that you have configured your monitor to enable an Adaptive Sync functionality.
2. Launch MonitorControl.
3. Go to the "Display" panel in the System Preferences.
4. Change the refresh rate setting to use Variable Refresh Rate.
5. Set the monitor to show the refresh rate using its OSD.
6. The refresh rate indicator reads 144Hz all the time, whatever application you use.

### Expected behavior

When MonitorControl is running, the Variable Refresh Rate setting in the OS is effective.

### Environment Information (please complete the following information)

```markdown
- macOS version: 12.3.1
- Mac model: Mac Studio (M1 Max 24 core GPU)
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): ASUS VG279
- Apple Silicon/M1 (yes or no): yes
```

**Comment by Hanage999** [comment]
One thing I forgot to mention: I connect my Mac and the monitor with a USB-C to 8K DisplayPort cable from Cable Matters.

**Comment by the0neyouseek** [comment]
Hi @Hanage999 
Apparently there are known bugs with VRR and Monterey. Please see this discussion https://github.com/MonitorControl/MonitorControl/discussions/981

**Comment by Hanage999** [comment]
Hi @the0neyouseek , thanks for the reply and guiding me to a relevant discussion.

In my case the opposite happens: I enable VRR in System Preferences, but the OS reverts the setting after sleeps, restarts or reconnections to its monitor.

Hopefully Apple will fix the crappy VRR support in one of the future Monterey updates.

**Comment by kolorfilm** [comment]
Same issue with my Dell G3223D. It doesn't store the VRR refresh rate after reboot. I'm on Monterey 12.5.1 (21G83). Seems to be still buggy there right?

**Reply by Hanage999** [reply]
FYI, it appears that the changes you make on the refresh rate is properly saved in Ventura (Beta 7), but VRR support is still glitchy.

**Comment by waydabber** [comment]
This does not seem to be a MonitorControl issue specifically. VRR might be affected if there is activity on-screen, MC does not create such activity unless there is an ongoing software brightness adjustment. For other issues like VRR setting not remembered etc. the problem seems to be entirely at macOS level.
